src/client/cli/cmd/delete.cpp

```diff
@@ -104,17 +104,40 @@ mp::ParseCode cmd::Delete::parse_args(mp::ArgParser* parser)
         return parse_code;
 
     request.set_purge(parser->isSet(purge_option));
+
+    bool instance_found = false, snapshot_found = false;
+    std::string instances, snapshots;
     for (const auto& item : add_instance_and_snapshot_names(parser))
     {
-        if (item.has_snapshot_name() && !request.purge())
+        if (!item.has_snapshot_name())
+        {
+            instances.append(fmt::format("{} ", item.instance_name()));
+            instance_found = true;
+        }
+        else
         {
-            cerr << "Snapshots can only be purged (after deletion, they cannot be recovered). Please use the `--purge` "
-                    "flag if that is what you want.\n";
-            return mp::ParseCode::CommandLineError;
+            snapshots.append(fmt::format("{}.{} ", item.instance_name(), item.snapshot_name()));
+            snapshot_found = true;
         }
 
         request.add_instances_snapshots()->CopyFrom(item);
     }
 
+    if (snapshot_found && !request.purge())
+    {
+        auto no_purge_base_error_msg =
+            "Snapshots can only be purged (after deletion, they cannot be recovered). Please use the `--purge` "
+            "flag if that is what you want";
+
+        if (instance_found)
+            cerr << fmt::format("{}:\n\n\tmultipass delete --purge {}\n\nYou can use a separate command to delete "
+                                "instances without purging them:\n\n\tmultipass delete {}\n",
+                                no_purge_base_error_msg, snapshots, instances);
+        else
+            cerr << fmt::format("{}.\n", no_purge_base_error_msg);
+
+        return mp::ParseCode::CommandLineError;
+    }
```

Comment:
Could this be extracted into a separate method/function, please?

Reply:
I moved the string to the unnamed namespace slimming this down. Does it make sense to create another function for just an `if` statement?

Reply:
> I moved the string to the unnamed namespace slimming this down

Thanks, that makes it better indeed.

> Does it make sense to create another function for just an if statement?

I think it definitely makes sense to have a function with only an `if` statement. The reader most likely wants to see something like `refuse_unpurged_snapshots(...)` without having to read 15 lines of details (which can always be dug into). In any case, here I was actually referring to the whole block that takes care of instance/snapshot arguments, including the `for` loop above. If it was me, I would probably extract that whole block into a function and then the `if` into another function.